### PR TITLE
Upload vehicle mod with working physics as per SMCE

### PR DESCRIPTION
**Pull request by adityak714**
The new uploaded car is a more real-life replication of a sedan (which was tried previously but could not hover in the emulator). It is now able to follow the physics of the SMCE default car, and is able to correctly move and turn in the terrain, with sensors working and intact.

### CODE REVIEW

**Comment by sh4r10** [comment]
How many and which sensors does this car have? Are they inline with what the previoius car had or have they been modififed?

**Comment by adityak714** [comment]
> How many and which sensors does this car have? Are they inline with what the previoius car had or have they been modififed?

The sensors are unchanged, and their position is at the same parts of the car. As the modded car's dimensions are now different in comparison to the square-like template car, they had to be repositioned very slightly.

**Comment by adityak714** [comment]
> Fix issue of env not registered + trying more sensors #15

This is a small follow-up commit to a previous problem that occurred, where the mod was not registered by the emulator. It is fixed, and I was also learning how to install more sensor placeholders in the mod, and that worked. Two new front IR sensors "Front Left Infrared"  and "Front Right Infrared" were made (as a trial). The Arduino sketch can be adjusted accordingly to also include them in the car's obstacle detection mechanism, potentially, or feedback can be given by the team if they need to be removed, replaced, tweaked, or more should be added.

File: mod/vehicle/PocketParkingVehicle.tscn [hunk 1]
```diff
@@ -178,6 +178,12 @@ transform = Transform( -4.37114e-08, 0, -1, 0, 1, 0, 1, 0, -4.37114e-08, 0.89230
 [node name="Back" type="Position3D" parent="AttachmentSlots"]
 transform = Transform( -1, 0, 8.74228e-08, 0, 1, 0, -8.74228e-08, 0, -1, 0, 0.4, 2.74527 )
 
+[node name="LeftFront" type="Position3D" parent="AttachmentSlots"]
+transform = Transform( 1, 0, 0, 0, 1, 0, 0, 0, 1, -1.00833, 0, -2.446 )
+
+[node name="RightFront" type="Position3D" parent="AttachmentSlots"]
+transform = Transform( 1, 0, 0, 0, 1, 0, 0, 0, 1, 0.980777, 0, -2.48073 )
+
```
Comment (adityak714) [inline mod/vehicle/PocketParkingVehicle.tscn:186]: Here are the placeholders for the two extra sensors at the front of the car, which were done as a trial.

File: mod/mods/rroverr.gd [hunk 2]
```diff
@@ -8,8 +8,8 @@ var mod_name: String = "example"
 
 # Called when the node enters the scene tree for the first time.
 func init(global) -> void:
-	global.register_environment("example/Example", preload("res://RestaurantSpicySalmon.tscn"))
-	global.register_vehicle("TemplateCar", preload("res://TemplateCar.tscn"))
+	global.register_environment("example/Example", preload("res://mods/PocketParkingTerrain.tscn"))
+	global.register_vehicle("PocketParkingVehicle", preload("res://vehicle/PocketParkingVehicle.tscn"))
```
Comment (adityak714) [inline mod/mods/rroverr.gd:12]: Some minor renaming of the terrain and vehicle mods

**Comment by adityak714** [comment]
In the mod directory of the repo, for the mod to be recognised by the emulator, one needs to find a file called "ppmod.pck" and copy-paste it in the user directory of the SMCE: 

For windows: C:/Users/adity/AppData/Roaming/Godot/app_userdata/SMCE
For Mac: ~/Library/Application Support/Godot/SMCE

**Review by Druner — APPROVED** [review]
looks good, car is sometimes a bit janky at specific higher speeds but it works fine, should prob write on the wiki on how to install the mod instead of a comment on the pull request

**Pull request merged**